File: src/Application Specific/Classes/Entity Collections/ContractsAlternative/ContractAlternativeSite/ContractAlternativeSiteValidator.cs
using System.Collections;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace ContractAlternativeSites
    {
        // make sure that contact object is valid
        public class ContractAlternativeSiteValidator : BaseValidator
        {
            public void Validate(ContractAlternativeSite oContractSite, ContractsAlternative.ContractAlternative contract)
            {

                // make sure that contact object is valid
                if (oContractSite.Errors.Count > 0)
                {
                    DictionaryEntry de;
                    foreach (var de in oContractSite.Errors)
                        AddCriticalMessage(Conversions.ToString(de.Value));
                }

                if (ValidatorMessages.CriticalMessages.Count > 0)
                {
                    throw new ValidationException(this);
                }
            }
        }
    }
}

File: src/Application Specific/Classes/Entity Collections/ContractsAlternative/ContractAlternativeSite/ContractAlternativeSiteValidator.vb
Namespace Entity

    Namespace ContractAlternativeSites

        Public Class ContractAlternativeSiteValidator
            'make sure that contact object is valid
            Inherits BaseValidator

            Public Sub Validate(ByVal oContractSite As ContractAlternativeSite, ByVal contract As Entity.ContractsAlternative.ContractAlternative)

                'make sure that contact object is valid
                If oContractSite.Errors.Count > 0 Then
                    Dim de As DictionaryEntry
                    For Each de In oContractSite.Errors
                        Me.AddCriticalMessage(CStr(de.Value))
                    Next
                End If


                If Me.ValidatorMessages.CriticalMessages.Count > 0 Then
                    Throw New ValidationException(Me)
                End If

            End Sub

        End Class

    End Namespace

End Namespace
